src/yaml/yamlUtils.ts

```diff
@@ -0,0 +1,257 @@
+    const endPositionForInsertion: Position = {...cursorPosition}
+
+    // Since there's no colon to separate the key and value, replace all text after the cursor.
+    endPositionForInsertion.character = currentLine.length
+
+    const lineOffsets = getLineOffsets(document.getText())
+
+    const trimmedLine = currentLine.trim()
+
+    if (trimmedLine.length === 0) {
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
+            document.getText().substring(0, currentLineEnd) +
+                '"":\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
+
+        tempPositionForCompletions.character += 1
+
+    // Trimmed line starts with '-'
+    } else if (trimmedLine.length >= 1 && trimmedLine[0] === '-') {
+        // Set start of insertion range to be immediately after the hyphen.
+        const indexOfHyphen = currentLine.indexOf('-')
+        startPositionForInsertion.character = indexOfHyphen + 1
+        shouldPrependSpace = true
+
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
+            document.getText().substring(0, currentLineEnd) +
+                (!currentLine.endsWith(' ') ? ' ' : '') +
+                ':\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
+
+    // Non-empty line but missing colon, add colon to end of current line
+    } else {
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
+            document.getText().substring(0, currentLineEnd) +
+                ':\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
+
+        // Starting pos is first non-space character
+        startPositionForInsertion.character = currentLine.indexOf(trimmedLine)
+    }
+
+    return {
+        modifiedDocText,
+        tempPositionForCompletions,
+        startPositionForInsertion,
+        endPositionForInsertion,
+        shouldPrependSpace
+    }
+}
+
+function processLineWithColon(document: TextDocument, cursorPosition: Position, currentLine: string, currentLineEnd: number): {
+    modifiedDocText: string,
+    tempPositionForCompletions: Position,
+    startPositionForInsertion: Position,
+    endPositionForInsertion: Position,
+    shouldPrependSpace: boolean
+} {
+    let modifiedDocText: string = document.getText()
+    let shouldPrependSpace: boolean = false
+
+    const tempPositionForCompletions: Position = {...cursorPosition}
+    const startPositionForInsertion: Position = {...cursorPosition}
+    const endPositionForInsertion: Position = {...cursorPosition}
+
+    const charNum = cursorPosition.character
+    const lineOffsets = getLineOffsets(document.getText())
+
+    // Current line has a colon, determine if cursor position is right or left of it
+    const colonIndex = currentLine.indexOf(':')
```

Comment:
here is a twist. A state name could actually contain colon. https://states-language.net/#states-fields . I think we need to account for it, both her as well as in the completions. Not sure, we might have to have quotes around string in this scenario.

Reply:
We'll have to think about this a little more. Will address this in the follow up commit.

Reply:
Thinking a little bit more about this situation, it's true that we can have colon inside state name, but there is no code completion that we provide on the line of the state name, so in my opinion we don't need to handle this case any differently.